lib/spack/spack/solver/asp.py

```diff
@@ -1130,6 +1135,19 @@ def __iter__(self):
         return iter(self.data)
 
 
+def parse_spec_from_yaml_string(string):
+    """Parse a spec from YAML and add file/line info to errors, if it's available."""
+    try:
+        spec = spack.spec.Spec(string)
+        return spec
+    except spack.parser.SpecSyntaxError as e:
+        mark = spack.config.get_mark_from_yaml_data(string)
+        if mark:
+            msg = f"{mark.name}:{mark.line + 1}: {str(e)}"
+            raise spack.parser.SpecSyntaxError(msg) from e
+        raise e
```

Comment:
This should be moved to a module dealing with YAML or config, seems out of place in the `spack.solver` package.

Reply:
moved to config